src/polymer/polymer-element.ts

```diff
@@ -239,53 +239,152 @@ function propertyToAttributeName(propertyName: string): string|null {
+  scannedElement.properties.forEach((o) => element.properties.push(o));
+  element.scriptElement = scannedElement.scriptElement;
+  scannedElement.slots.forEach((o) => element.slots.push(o));
+  element.sourceRange = scannedElement.sourceRange!;
+  element.superClass = scannedElement.superClass;
+  element.tagName = scannedElement.tagName;
+  scannedElement.warnings.forEach((o) => element.warnings.push(o));
+
+  //
+  // Behaviors
+  //
+  const behaviorsAndWarnings =
+      getBehaviors(scannedElement.behaviorAssignments, document);
 
   // This has the combined effects of copying the array of warnings from the
   // ScannedElement, and adding in any new ones found when resolving behaviors.
-  clone.warnings = clone.warnings.concat(flatteningResult.warnings);
-
-  const behaviors = Array.from(flatteningResult.resolvedBehaviors);
-  clone.properties = mergeByName(
-      scannedElement.properties,
-      behaviors.map((b) => ({name: b.className, vals: b.properties})));
-  clone.attributes = mergeByName(
-      scannedElement.attributes,
-      behaviors.map((b) => ({name: b.className, vals: b.attributes})));
-  clone.events = mergeByName(
-      scannedElement.events,
-      behaviors.map((b) => ({name: b.className, vals: b.events})));
+  element.warnings = element.warnings.concat(behaviorsAndWarnings.warnings);
+
+  const behaviors = Array.from(behaviorsAndWarnings.behaviors);
+
+  // for (const behavior of behaviors) {
+  //   inheritFrom(element, behavior);
+  // }
+
+  element.properties = inheritValues(
+      element.properties,
+      behaviors.map((b) => ({source: b.className, values: b.properties})));
+  element.attributes = inheritValues(
+      element.attributes,
+      behaviors.map((b) => ({source: b.className, values: b.attributes})));
+  element.events = inheritValues(
+      element.events,
+      behaviors.map((b) => ({source: b.className, values: b.events})));
 
   const domModule = document.getOnlyAtId(
       'dom-module',
       scannedElement.tagName || '',
       {imported: true, externalPackages: true});
+
   if (domModule) {
-    clone.description = scannedElement.description || domModule.comment || '';
-    clone.domModule = domModule.node;
-    clone.slots = domModule.slots.slice();
-    clone.localIds = domModule.localIds.slice();
+    element.description = element.description || domModule.comment || '';
+    element.domModule = domModule.node;
+    element.slots = domModule.slots.slice();
+    element.localIds = domModule.localIds.slice();
   }
 
   if (scannedElement.pseudo) {
-    clone.kinds.add('pseudo-element');
+    element.kinds.add('pseudo-element');
+  }
+
+  return element;
+}
+
+/**
+ * Note: mutates `element`.
+ */
+function inheritFrom(element: PolymerElement, superElement: PolymerElement) {
+  // TODO(justinfagnani): fixup and use inheritValues, but it has slightly odd
+  // semantics currently
+
+  for (const superProperty of superElement.properties) {
+    // TODO(justinfagnani): better clone/merge
```

Comment:
I was referring to a deep clone/merge. But given that overrides completely shadow the overridden, it's not necessary. Behaviors do need to merge though - they currently don't :(